generator.go

```diff
@@ -272,92 +260,59 @@ func (g *Gen) NewV4() (UUID, error) {
 	u := newFromHash(sha1.New(), ns, name)
 	u.SetVersion(V5)
-	u.SetVariant(VariantRFC4122)
+	u.SetVariant(VariantRFC9562)
 
 	return u
 }
 
 // NewV6 returns a k-sortable UUID based on a timestamp and 48 bits of
 // pseudorandom data. The timestamp in a V6 UUID is the same as V1, with the bit
 // order being adjusted to allow the UUID to be k-sortable.
-//
-// This is implemented based on revision 03 of the Peabody UUID draft, and may
-// be subject to change pending further revisions. Until the final specification
-// revision is finished, changes required to implement updates to the spec will
-// not be considered a breaking change. They will happen as a minor version
-// releases until the spec is final.
 func (g *Gen) NewV6() (UUID, error) {
+	/* https://datatracker.ietf.org/doc/html/rfc9562#name-uuid-version-6
+	    0                   1                   2                   3
+	    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+	   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	   |                           time_high                           |
+	   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	   |           time_mid            |  ver  |       time_low        |
+	   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	   |var|         clock_seq         |             node              |
+	   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+	   |                              node                             |
+	   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ */
 	var u UUID
 
-	if _, err := io.ReadFull(g.rand, u[10:]); err != nil {
-		return Nil, err
-	}
-
-	timeNow, clockSeq, err := g.getClockSequence(false)
+	timeNow, _, err := g.getClockSequence(false)
 	if err != nil {
 		return Nil, err
 	}
 
 	binary.BigEndian.PutUint32(u[0:], uint32(timeNow>>28))   // set time_high
 	binary.BigEndian.PutUint16(u[4:], uint16(timeNow>>12))   // set time_mid
 	binary.BigEndian.PutUint16(u[6:], uint16(timeNow&0xfff)) // set time_low (minus four version bits)
-	binary.BigEndian.PutUint16(u[8:], clockSeq&0x3fff)       // set clk_seq_hi_res (minus two variant bits)
-
-	u.SetVersion(V6)
-	u.SetVariant(VariantRFC4122)
-
-	return u, nil
-}
 
-// getClockSequence returns the epoch and clock sequence for V1,V6 and V7 UUIDs.
-//
-//	When useUnixTSMs is false, it uses the Coordinated Universal Time (UTC) as a count of 100-
-//
-// nanosecond intervals since 00:00:00.00, 15 October 1582 (the date of Gregorian reform to the Christian calendar).
-func (g *Gen) getClockSequence(useUnixTSMs bool) (uint64, uint16, error) {
-	var err error
-	g.clockSequenceOnce.Do(func() {
-		buf := make([]byte, 2)
-		if _, err = io.ReadFull(g.rand, buf); err != nil {
-			return
-		}
-		g.clockSequence = binary.BigEndian.Uint16(buf)
-	})
-	if err != nil {
-		return 0, 0, err
+	//Based on the RFC 9562 recommendation that this data be fully random and not a monotonic counter,
```

Comment:
Nit: comments that aren't directives should have a space after `//`

Reply:
```suggestion
	// Based on the RFC 9562 recommendation that this data be fully random and not a monotonic counter,
```